### PR TITLE
Update Testing.md with pre-processing information

### PR DESCRIPTION
**Pull request by ericvera**
With the added information a react-native developer will be able to run jest tests without errors. First time contributing to react-native so please let me know if I missed something :)

As it is I was running into issues running jest tests. The configuration added was added from the react-native package.json.

### CODE REVIEW

File: docs/Testing.md [hunk 1]
```diff
@@ -25,6 +25,33 @@ npm test
 
 from the react-native root, and we encourage you to add your own tests for any components you want to contribute to.  See [`getImageSource-test.js`](https://github.com/facebook/react-native/blob/master/Examples/Movies/__tests__/getImageSource-test.js) for a basic example.
 
+Note: In order to run your own tests, you will have to first follow the Getting Started instructions on the Jest page and then include the `jest` objects below in `packages.json` so that the scripts are pre-processed before execution.
```
Comment (ide) [inline docs/Testing.md]: packages.json -> package.json

File: docs/Testing.md [hunk 2]
```diff
@@ -25,6 +25,33 @@ npm test
...
+  "testPathIgnorePatterns": [
+    "/node_modules/",
+    "packager/react-packager/src/Activity/"
+    ],
+    "testFileExtensions": [
+      "js"
+    ],
+    "unmockedModulePathPatterns": [
+      "promise",
+      "source-map"
```
Comment (ide) [inline docs/Testing.md]: dedent lines 43-49

**Comment by ericvera** [comment]
Feedback addressed and commits squashed.

**Comment by ide** [comment]
Thanks!

**Comment by brentvatne** [comment]
This definitely does not work with `require('react-native')` :smile:

**Comment by ericvera** [comment]
Those settings unblocked me from running tests of libraries (yes those did not rely on react-native). The one time I tried tests that included require('react-native') it did not work. Possibly the "mock-by-default" behavior?

**Pull request merged**